modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -300,11 +306,28 @@ void ONNXImporter::addLayer(Net& dstNet, LayerParams& layerParams,
     }
 }
 
+void addConstantNodesForInitializers(opencv_onnx::GraphProto& graph_proto)
+{
+    int num_initializers = graph_proto.initializer_size();
+    for (int id = 0; id < num_initializers; id++)
+    {
+        opencv_onnx::TensorProto initializer = graph_proto.initializer(id);
+        opencv_onnx::NodeProto* constant_node = graph_proto.add_node();
+        constant_node->set_op_type("Constant");
+        constant_node->set_name(initializer.name());
+        constant_node->add_output(initializer.name());
+        opencv_onnx::AttributeProto* value = constant_node->add_attribute();
+        opencv_onnx::TensorProto* tensor = value->add_tensors();
+        tensor->CopyFrom(initializer);
```

Comment:
Let's also remove initialized nodes in the loop after they copied to the constant node

Reply:
If I remove intializers inside this loop. then this function would be useless because it wont be able to fetch initializer data.
https://github.com/opencv/opencv/blob/c9c60783df11f4434d9b6afcd3ff00f5518682a7/modules/dnn/src/onnx/onnx_importer.cpp#L133-L147
which is being called here
https://github.com/opencv/opencv/blob/c9c60783df11f4434d9b6afcd3ff00f5518682a7/modules/dnn/src/onnx/onnx_importer.cpp#L330-L334

So we will have to fetch data from `constant` nodes, means we need to create a function which first go through all of the nodes and fetch data from all `constant` nodes. 
Am I right ?

Reply:
Let's see, it `set_allocated_t` will work - we can keep as is.

Reply:
@dkurt , please have a look at the changes.

Reply:
Setting tensor directly from `initializer` wasn't working
`value->set_allocated_t(initializer);`
It was giving me Segmentation core error so I created a new tensor and set that to value attribute. So it is copied tensor. 
```
opencv_onnx::TensorProto* tensor = initializer.New();
tensor->CopyFrom(initializer);
```

Reply:
Can you experiment if we get `mutable_initializer` and then set it without copying by `set_allocated_t` everything will be fine or it will crash? It depends on ownership policy in case of `set_allocated_t` - will protobuf release the object twice or not.

Reply:
I have already tried it. Its not working.

Reply:
Got it, thanks!
So can you try now right after `tensor->CopyFrom(initializer);` clear the origin tensor and remove it from the graph? For memory optimization.

Reply:
```
tensor->CopyFrom(initializer);
releaseONNXTensor(initializer);
value->set_allocated_t(tensor);
```

where releaseONNXTensor is
```
void releaseONNXTensor(opencv_onnx::TensorProto& tensor_proto)
{
    if (!tensor_proto.raw_data().empty()) {
        delete tensor_proto.release_raw_data();
    }
}
```
I think you meant the same.

Reply:
Something like that but we have no raw_data in this case

Reply:
all the initializers have data type as raw_data. I have checked data types. 
```
bias: raw data

running_mean: raw data

running_var: raw data

weight: raw data

```
Do you think we should modify this function for all data_types or a new function should be created to check and delete all data_types and let this function as it is ?

Reply:
No, that's not necessary. We can just call method `clear_initializer`.

Reply:
i think we cannot use
`graph_proto.clear_initializer()` before calling `getGraphTensors(graph_proto)`.